monai/metrics/occlusion_sensitivity.py

```diff
@@ -0,0 +1,188 @@
+    be useful to understand why a network is making certain decisions.
+
+    The result is given as ``baseline`` (the probability of
+    a certain output) minus the probability of the output with the occluded
+    area.
+
+    Therefore, higher values in the output image mean there was a
+    greater the drop in certainty, indicating the occluded region was more
+    important in the decision process.
+
+    See: R. R. Selvaraju et al. Grad-CAM: Visual Explanations from Deep Networks via
+    Gradient-based Localization. https://doi.org/10.1109/ICCV.2017.74
+
+    Args:
+        model: classification model to use for inference
+        image: image to test. Should be tensor consisting of 1 batch, can be 2- or 3D.
+        label: classification label to check for changes (normally the true
+            label, but doesn't have to be)
+        pad_val: when occluding part of the image, which values should we put
+            in the image?
+        margin: we'll create a cuboid/cube around the voxel to be occluded. if
+            ``margin==2``, then we'll create a cube that is +/- 2 voxels in
+            all directions (i.e., a cube of 5 x 5 x 5 voxels). A ``Sequence``
+            can be supplied to have a margin of different sizes (i.e., create
+            a cuboid).
+        n_batch: number of images in a batch before inference.
+        b_box: Bounding box on which to perform the analysis. The output image
+            will also match in size. There should be a minimum and maximum for
+            all dimensions except batch: ``[min1, max1, min2, max2,...]``.
+            * By default, the whole image will be used. Decreasing the size will
+            speed the analysis up, which might be useful for larger images.
+            * Min and max are inclusive, so [0, 63, ...] will have size (64, ...).
+            * Use -ve to use 0 for min values and im.shape[x]-1 for xth dimension.
+    Returns:
+        Numpy array. If no bounding box is supplied, this will be the same size
+        as the input image. If a bounding box is used, the output image will be
+        cropped to this size.
+    """
+
+    # Check input arguments
+    image = _check_input_image(image)
+    label = _check_input_label(label, image)
+    im_shape = np.array(image.shape[1:])
+    b_box_min, b_box_max = _check_input_bounding_box(b_box, im_shape)
+
+    # Get baseline probability
+    baseline = model(image).detach()[0, label].item()
+
+    # Create some lists
+    batch_images = []
+    batch_ids = []
+
+    sensitivity_im = torch.empty(0, dtype=torch.float32, device=image.device)
+
+    # If no bounding box supplied, output shape is same as input shape.
+    # If bounding box is present, shape is max - min + 1
+    output_im_shape = im_shape if b_box is None else b_box_max - b_box_min + 1
+    num_required_predictions = np.prod(output_im_shape)
+
+    # Loop 1D over image
+    for i in trange(num_required_predictions):
```

Comment:
looks like there's some memory issue with the Windows test, I don't know the root cause for now... is it possible that this for loop is actually accumulating the model variable gradients, using a lot of memory? adding some `detach()` commands here might help?

Reply:
The Windows test was originally failing but I decreased batch size and that seems to be passing now [link](https://github.com/Project-MONAI/MONAI/pull/1099/checks?check_run_id=1283108651#step:9:2582). The current error on Windows looks unrelated. I'll relaunch and see what happens.